File: content/LLD/Reference-Code/3. Memento Pattern/main.py
from history import History
from text_editor import TextEditor
from text_memento import TextMemento

text_editor = TextEditor()
history = History()

text_editor.write("Hello")
text_editor.write(" World")
history.save_state(text_editor.save())
text_editor.write(" Good")
text_editor.write(" Bye")
history.save_state(text_editor.save())
print(text_editor.get_text())
print("-------")
text_editor.restore(history.undo())
print(text_editor.get_text())
text_editor.restore(history.undo())
print(text_editor.get_text())

File: content/LLD/Reference-Code/3. Memento Pattern/Main.java
public class Main {
    public static void main(String[] args) {
        TextEditor textEditor = new TextEditor();
        History history = new History();

        textEditor.write("Hello");
        textEditor.write(" World");
        history.saveState(textEditor.save());
        textEditor.write(" Good");
        textEditor.write(" Bye");
        history.saveState(textEditor.save());
        System.out.println(textEditor.getText());
        System.out.println("-------");
        textEditor.restore(history.undo());
        System.out.println(textEditor.getText());
        textEditor.restore(history.undo());
        System.out.println(textEditor.getText());
    }
}
